apps/angular/src/app/features/anime-table/anime-table.component.ts

```diff
@@ -16,19 +34,64 @@ enum ColumnKey {
 	Status = 'status',
 }
 
+const INITIAL_PAGE_SIZE = 25;
+
+const DEBOUNCE_TIME = 500;
+
 /** Anime table component. */
 @Component({
 	selector: 'camp-anime-table',
 	templateUrl: './anime-table.component.html',
 	styleUrl: './anime-table.component.css',
 	standalone: true,
-	imports: [MatTableModule, AsyncPipe, DatePipe, EmptyPipe],
+	imports: [
+		MatTableModule,
+		MatPaginatorModule,
+		DatePipe,
+		EmptyPipe,
+		MatSortModule,
+		SearchFormComponent,
+		AnimeTypeFilterComponent,
+		AsyncPipe,
+	],
 })
-export class AnimeTableComponent {
+export class AnimeTableComponent implements OnInit, AfterViewInit, OnChanges {
+	/** Search value. */
+	@Input()
+	public search = '';
+
+	/** Filter value. */
+	@Input()
+	public filter: readonly AnimeType[] = [];
+
 	private readonly animeService = inject(AnimeService);
 
-	/** Represents anime list. */
-	protected readonly anime$ = this.animeService.getAll();
+	private readonly activatedRoute = inject(ActivatedRoute);
+
+	private readonly navigationService = inject(NavigationService);
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	private readonly pageNumber$ = new BehaviorSubject(0);
+
+	private readonly pageSize$ = new BehaviorSubject(INITIAL_PAGE_SIZE);
+
+	private readonly ordering$ = new BehaviorSubject<string | null>(null);
+
+	private readonly search$ = new BehaviorSubject('');
+
+	/** Has fetching error. */
+	protected readonly hasFetchingError$ = new BehaviorSubject(false);
+
+	/** Search value emitter. */
+	@Output()
+	public readonly searchValueEmitter = new EventEmitter<string>();
+
+	/** Filter value emitter. */
+	@Output()
+	public readonly filterValueEmitter = new EventEmitter<readonly AnimeType[]>();
+
+	private readonly filter$ = new BehaviorSubject<readonly AnimeType[]>([]);
```

Comment:
'search' is also a filter. Maybe adjust naming of the 'filter' property to be more clear. Something like this:
```suggestion
	private readonly animeTypesFilter$ = new BehaviorSubject<readonly AnimeType[]>([]);
```